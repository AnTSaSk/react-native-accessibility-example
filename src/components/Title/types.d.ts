import { themeStyleProps } from '../../hooks/types.d';

export interface TitleProps {
  label: string;
  isDark?: boolean;
  theme?: themeStyleProps;
}
