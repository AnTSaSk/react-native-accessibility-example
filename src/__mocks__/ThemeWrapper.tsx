import React, { FunctionComponent } from 'react';
import { ThemeProvider } from '@emotion/react';

import { lightColors } from '../variables';

const ThemeWrapper: FunctionComponent<{}> = ({ children }) => (
  <ThemeProvider theme={{ colors: lightColors }}>{children}</ThemeProvider>
);

export default ThemeWrapper;
