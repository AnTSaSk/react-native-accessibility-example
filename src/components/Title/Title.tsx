import React, { FunctionComponent } from 'react';
import { Text } from 'react-native';

// Types
import { TitleProps } from './types.d';

const Title: FunctionComponent<TitleProps> = ({ label }) => (
  <Text
    // style={[styles.sectionTitle, { color: themeColors.colors.title }]}
    accessible
    accessibilityLabel={label}
    accessibilityRole="text"
  >
    {label}
  </Text>
);

export default Title;
