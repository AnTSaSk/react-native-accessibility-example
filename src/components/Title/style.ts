import styled from '@emotion/native';

import { TitleProps } from './types';

export const TitleComponent = styled.Text<TitleProps>`
  margin: 0 0 15px;
  color: ${(props) =>
    props.isDark ? props.theme.colors.white : props.theme.colors.grey};
  font-size: 26px;
`;
