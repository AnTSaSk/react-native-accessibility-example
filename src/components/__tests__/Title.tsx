import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

// Mocked context
import ThemeWrapper from '../../__mocks__/ThemeWrapper';

// Component
import Title from '../Title';

const TestComponent = (
  <ThemeWrapper>
    <Title label="Test content" />
  </ThemeWrapper>
);

it('Renders correctly Title component', () => {
  const title = renderer.create(TestComponent).toJSON();

  expect(title).toMatchSnapshot();
});
