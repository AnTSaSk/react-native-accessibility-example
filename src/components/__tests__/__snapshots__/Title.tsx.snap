// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Title component 1`] = `
<Text
  accessibilityLabel="Test content"
  accessibilityRole="text"
  accessible={true}
  isDark={false}
  label="Test content"
  style={
    Object {
      "color": "#A9A9A9",
      "fontSize": 26,
      "marginBottom": 15,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
    }
  }
>
  Test content
</Text>
`;
